db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1551,13 +1551,21 @@ Status DBImpl::FlushMemTable(ColumnFamilyData* cfd,
     if (!cfd->mem()->IsEmpty() || !cached_recoverable_state_empty_.load()) {
       s = SwitchMemtable(cfd, &context);
     }
-
     if (s.ok()) {
       if (cfd->imm()->NumNotFlushed() != 0 || !cfd->mem()->IsEmpty() ||
           !cached_recoverable_state_empty_.load()) {
         flush_memtable_id = cfd->imm()->GetLatestMemTableID();
         flush_req.emplace_back(cfd, flush_memtable_id);
       }
+      if (cfd->GetName().compare(kPersistentStatsColumnFamilyName) != 0 &&
+          immutable_db_options_.persist_stats_to_disk) {
+        ColumnFamilyData* cfd_stats =
+            versions_->GetColumnFamilySet()->GetColumnFamily(
+                kPersistentStatsColumnFamilyName);
+        s = SwitchMemtable(cfd_stats, &context);
```

Comment:
Just to understand: are we flushing stats CF every time any CF is flushed?

Reply:
Yes, actually more than that, `FlushMemTable` can be called by the following: 
```
GetLiveFiles,
CompactRange,
Flush,
ResumeImpl,
IngestExternalFiles
```
Shall we limit the change to manual flush only?

Reply:
What's the necessity of doing that? If it doesn't block WAL from being purged, do we still need to flush it?

Reply:
added check to only force flushing stats cf when it will be the only CF lagging behind after the current flush